tekstowo.py

```diff
@@ -0,0 +1,464 @@
+            print(e)
+
+        print(f"{ARTIST_LETTER}: Visited {page}/{limit}")
+       
+    print(f"{generate_timestamp()}: Letter {ARTIST_LETTER}: collected {len(urls)} artists")
+    print(f"Letter {ARTIST_LETTER} artists collected.")
+
+    return urls, len(urls)
+
+def get_artist_songs(artist_url: str) -> list:
+    """
+    Extract all the songs from a given artist.
+    """
+    urls = []
+    processed_first_page = False
+
+    while not processed_first_page:
+        time.sleep(5)
+        try:
+            response = requests.get(artist_url, timeout=60)
+
+            # Check for request success
+            response.raise_for_status()
+
+            soup = BeautifulSoup(response.content, 'html.parser')
+            songs = soup.find_all(class_='box-przeboje')
+            artist = soup.find(class_="col-md-7 col-lg-8 px-0")
+            artist = artist.text.split(" (")[0].strip()
+
+            for song in songs:
+                song_title_element = song.find(class_='title')
+
+                if song_title_element:
+                    if artist in song_title_element.text.strip():
+                        song_url = "https://tekstowo.pl" + song_title_element['href']
+                        if not ".plpiosenka" in song_url and song_url not in urls and not "dodaj_tekst" in song_url:
+                            urls.append(song_url)
+
+            button_next_page = soup.find_all(class_='page-link')
+            if button_next_page and len(button_next_page) > 0:
+                button_next_page = button_next_page[-1]
+
+                if 'następna' in button_next_page.text.lower():
+                    artist_url = "https://tekstowo.pl" + button_next_page['href']
+                else:
+                    break
+            else:
+                break
+
+        except requests.exceptions.RequestException as e:
+            print(f"An error occurred during the HTTP request: {e}")
+            return False
+
+        except Exception as e:
+            print(f"An unexpected error occurred: {e}")
+            return False
+    
+    print(f"Artist {artist}, collected {len(urls)} song URLs")
+    return urls
+
+def extract_song(song_url: str) -> Tuple[str, str, str]:
+    """
+    Scrapes the song (and song_translation if found) and its
+    title from a given song_url.
+    """
+
+    song = ""
+    song_translation = None
+
+    try:
+        response = requests.get(song_url, timeout=60)
+        if response.ok:
+            # Scrape URL content with BeautifulSoup
+            soup = BeautifulSoup(response.content, 'lxml')
+
+            # Song title
+            song_title = soup.find(class_='col-lg-7').text.strip()
+      
+            # Original song
+            song_html = soup.find(class_='inner-text')			
+            song = song_html.text.strip()
+
+            # Translated version
+            transl_html = soup.find('div', id='translation')
+            song_translation = transl_html.text.strip().split("\t\t")[0]
+
```

Comment:
A single space isn't need in the try-except block of code as it is one, continuous 'block entity'.

Reply:
Changed to no space in try-except block.